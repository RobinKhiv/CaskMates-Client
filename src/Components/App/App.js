import React, { Component } from 'react';
import {Route,Switch} from 'react-router-dom';
import Navigation from '../Navigation/Navigation';
import LandingPage from '../../Routes/LandingPage';
import AddWhiskeyPage from '../../Routes/AddWhiskeyPage';
import ReviewWhiskeyPage from '../../Routes/ReviewWhiskeyPage';
import LoginPage from '../../Routes/LoginPage';
import ListPage from '../../Routes/ListPage';
import WhiskeyLookupPage from '../../Routes/WhiskeyLookupPage';
import WhiskeyPage from '../../Routes/WhiskeyPage';
import RegistrationPage from '../../Routes/RegistrationPage';
import NotFoundPage from '../../Routes/NotFoundPage';
import PrivateRoute from '../../Components/Utils/PrivateRoute'
import PublicOnlyRoute from '../../Components/Utils/PublicOnlyRoute'
import WhiskeyContext from '../../Context/WhiskeyContext';
import './App.css';

class App extends Component {
  static contextType = WhiskeyContext;

  render() {
    return (
      <React.Fragment>
        <Navigation />
        <main role='main'>
          <Switch>
            <Route exact path={'/'} component={LandingPage} />
            <Route path={'/whiskeyList'} component={ListPage} />
            <Route exact path={'/whiskeys'} component={WhiskeyLookupPage}/>
            <Route exact path={'/whiskeys/:whiskeyId'} component={WhiskeyPage}/>
            <PrivateRoute path={'/whiskeys/:whiskeyId/addReview'} component={ReviewWhiskeyPage}/>
            <PrivateRoute path={'/addWhiskey'} component={AddWhiskeyPage} />
            <PublicOnlyRoute path={'/login'} component={LoginPage} />
            <PublicOnlyRoute path={'/register'} component={RegistrationPage} />
            <Route component={NotFoundPage} />
          </Switch>
        </main>
      </React.Fragment>
    );
  }
}

export default App;
