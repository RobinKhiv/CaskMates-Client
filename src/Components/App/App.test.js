import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import {BrowserRouter} from 'react-router-dom';

describe('<App />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <App />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

  it('renders without crashing', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot(); 
  });
});