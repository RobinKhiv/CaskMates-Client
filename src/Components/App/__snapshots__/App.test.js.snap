// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
Array [
  <nav
    className="navigation"
    role="navigation"
  >
    <a
      className="brand"
      href="/"
      onClick={[Function]}
    >
      Caskmates
    </a>
    <div
      className="navbar-anchors"
    >
      <a
        className="nav-a"
        href="/whiskeys"
        onClick={[Function]}
      >
        Whiskey Lookup
      </a>
      <span>
        |
      </span>
      
      <div
        className="Header__not-logged-in"
      >
        <a
          className="nav-a log"
          href="/login"
          onClick={[Function]}
        >
          <img
            alt="login icon"
            height="14"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSrdZKzkouFPEorhZMdksNlgtVyANA_0YJl1NYGqlpaELG4TslJig"
          />
        </a>
      </div>
    </div>
  </nav>,
  <main
    role="main"
  >
    <section
      className="row"
    >
      <div
        className="landing-page-container col-6"
      >
        <h3
          className="Logo"
        >
          Caskmates
        </h3>
        <p
          className="branding-quote"
        >
          For us whiskey isn't just a drink, its a passion!
        </p>
        <p>
           This is your one stop for all the info you'll ever need on whiskey needs. You'll be able to construct your own personalized list with your favorite drink or keep track of new whiskeys you can not wait to try. simply share your expertise so others can see what they are missing out!
        </p>
        <a
          href="/login"
          onClick={[Function]}
        >
          <input
            name="get-started-btn"
            type="button"
            value="Get Started"
          />
        </a>
      </div>
    </section>
  </main>,
]
`;
